coupler.cpp

```diff
@@ -2,6 +2,8 @@
 
 #include <chrono>
 #include <thread>
+#include <filesystem>
```

Comment:
this is non-standard c++14. Any alternatives?

Reply:
I actually think the other PR resolves the same thing while being C++14-compliant. 

Though IIRC this C++17 version is technically more correct, since opening an ifstream can fail for more reasons than the file not existing.

We should just be able to close this one, and merge the other branch, then I can update the geos rocm branch submodule appropriately.